File: UVA/p11498.py
while True:
    n = int(input())
    if n == 0:
        break
    x,y = map(int,input().split())
    for i in range(n):
        xx,yy = map(int,input().split())
        if xx == x or yy == y:
            print('divisa')
        elif xx>x and yy>y:
            print('NE')
        elif xx>x and yy<y:
            print('SE')
        elif xx<x and yy>y:
            print('NO')
        else:
            print('SO')

File: UVA/p11498.java

import java.util.Scanner;

/**
 * AC, easy, ad hoc, 29 August 2018.
 * @author Faiz Ikhwan
 */
public class p11498 
{
    public static void main(String[] args)
    {
        Scanner input = new Scanner(System.in);
        int inp;
        
        while(true)
        {
            inp = input.nextInt();
            if(inp == 0)
                break;                        
            else
            {               
                int divX = input.nextInt();
                int divY = input.nextInt();
                
                for(int i=0; i<inp; i++)
                {
                    int x = input.nextInt();
                    int y = input.nextInt();
                    
                    if(x == divX || y == divY)
                        System.out.println("divisa");
                    else if(x > divX && y > divY)
                        System.out.println("NE");
                    else if(x > divX && y < divY)
                        System.out.println("SE");
                    else if(x < divX && y > divY)
                        System.out.println("NO");
                    else
                        System.out.println("SO");
                }
            }
        }
    }
}
